gh-125436: add missing allow_unnamed_section doc; split too long line

litlighilit opened the pull request:
<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
gh-NNNNN: Summary of the changes made
```

Where: gh-NNNNN refers to the GitHub issue number.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `main`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `main`.

-->


<!-- gh-issue-number: gh-125436 -->
* Issue: gh-125436
<!-- /gh-issue-number -->


<!-- readthedocs-preview cpython-previews start -->
----
📚 Documentation preview 📚: https://cpython-previews--125437.org.readthedocs.build/

<!-- readthedocs-preview cpython-previews end -->

litlighilit committed:
gh-125436: add missing allow_unnamed_section doc

litlighilit committed:
gh-125436: split too long line of ConfigParser doc

Eclips4 reviewed (changes requested):
Please add a ``versionchanged`` section for the ``allow_unnamed_section`` parameter.

litlighilit committed:
gh-125436: fixup missing desc and verionchanged

vadmium commented on Doc/library/configparser.rst:
```diff
@@ -942,7 +942,13 @@ interpolation if an option used is not defined elsewhere. ::
 
-.. class:: ConfigParser(defaults=None, dict_type=dict, allow_no_value=False, delimiters=('=', ':'), comment_prefixes=('#', ';'), inline_comment_prefixes=None, strict=True, empty_lines_in_values=True, default_section=configparser.DEFAULTSECT, interpolation=BasicInterpolation(), converters={})
+.. class:: ConfigParser(defaults=None, dict_type=dict, allow_no_value=False, \
+                        delimiters=('=', ':'), comment_prefixes=('#', ';'), \
```
I think there should be an asterisk * item, because _allow_unnamed_section_, and all other parameters from _delimiters_, are keyword-only.
```suggestion
.. class:: ConfigParser(defaults=None, dict_type=dict, allow_no_value=False, *, \
                        delimiters=('=', ':'), comment_prefixes=('#', ';'), \
```

vadmium commented on Doc/library/configparser.rst:
```diff
@@ -990,6 +996,13 @@ ConfigParser Objects
+   specified ahead (often called ``"Global Options"`` or
+   ``options in "Default Section"``) are allowed. See
+   `"Supported INI File Structure" section<#supported-ini-file-structure>`_
+   for details.
```
I think drop the last part of the first sentence. Nowhere else mentions _global options_, and _default section_ is a different thing.

Suggest linking directly to the Unnamed Sections section. Maybe spacing the link target from its text might fix the compilation error too.

```suggestion
   When *allow_unnamed_section* is ``True`` (default: ``False``),
   the first section name can be omitted. See the
   `"Unnamed Sections" section <#unnamed-sections>`.
```

litlighilit replied:
Thanks!

The only fly in the ointment is the missing `_` to after `<#unnamed-sections>`

litlighilit committed:
gh-125436: add `*`; apply suggestion from @picnixz

- add space in ref before `<` char of URL to fix compile error;
- add `*` to mark keywords (fixup) of ConfigParser;
- simpify desc for `allow_unnamed_section` param;

Eclips4 reviewed (changes requested):
Thanks for the changes!
Taking a closer look at the #117273 I've noticed that the class `ConfigParser` inherits from the `RawConfigParser` and therefore parameter `allow_unnamed_section` has actually been added to the `RawConfigParser` class. Therefore, we need to do two things:
* Update doc for `ConfigParser` (which you have already done!)
* Update doc for `RawConfigParser`. Although this class is considered legacy in docs, it would still be useful to update its signature. The section with an explanation what `allow_unnamed_section` is doing unnecessary here.

litlighilit committed:
gh-125436: update signature of RawConfigParser...

- apply suggestion from https://github.com/python/cpython/pull/125437#pullrequestreview-2375551319
- also update for  `versionchanged` for params of RawConfigParser

Eclips4 committed:
Merge branch 'main' into lit-fix-doc

Eclips4 reviewed (approved):
Sorry for waiting @litlighilit! LGTM. Thank you!
